zhaquirks/tuya/tuya_siren.py

```diff
@@ -98,3 +203,255 @@ class TuyaSirenRingtone(t.enum8):
+    .applies_to("_TZE204_q76rtoa9", "TS0601")
+    .tuya_enum(
+        dp_id=5,
+        attribute_name="alarm_volume",
+        enum_class=NeoAlarmVolume,
+        translation_key="alarm_volume",
+        fallback_name="Alarm volume",
+    )
+    .tuya_number(
+        dp_id=7,
+        attribute_name="alarm_duration",
+        min_value=0,
+        type=t.uint16_t,
+        max_value=1800,
+        step=1,
+        unit=UnitOfTime.SECONDS,
+        translation_key="alarm_duration",
+        fallback_name="Alarm duration",
+    )
+    .tuya_switch(
+        dp_id=13,
+        attribute_name="siren_on",
+        entity_type=EntityType.STANDARD,
+        translation_key="siren_on",
+        fallback_name="Siren on",
+    )
+    .tuya_battery(
+        dp_id=15, battery_type=BatterySize.Other, battery_qty=1, battery_voltage=30
+    )
+    .tuya_enum(
+        dp_id=21,
+        attribute_name="alarm_ringtone",
+        enum_class=NeoAlarmMelody18,
+        translation_key="alarm_ringtone",
+        fallback_name="Alarm ringtone",
+    )
+    .skip_configuration()
+    .add_to_registry()
+)
+
+
+# Tuya ZA03
+(
+    TuyaQuirkBuilder("_TZE200_t1blo2bj", "TS0601")
+    .applies_to("_TZE204_t1blo2bj", "TS0601")
+    .applies_to("_TZE204_q76rtoa9", "TS0601")
+    .tuya_enum(
+        dp_id=5,
+        attribute_name="alarm_volume",
+        enum_class=TuyaAlarmVolume,
+        translation_key="alarm_volume",
+        fallback_name="Alarm volume",
+    )
+    .tuya_number(
+        dp_id=7,
+        attribute_name="alarm_duration",
+        min_value=0,
+        type=t.uint16_t,
+        max_value=380,
+        step=1,
+        unit=UnitOfTime.SECONDS,
+        translation_key="alarm_duration",
+        fallback_name="Alarm duration",
+    )
+    .tuya_switch(
+        dp_id=13,
+        attribute_name="siren_on",
+        entity_type=EntityType.STANDARD,
+        translation_key="siren_on",
+        fallback_name="Siren on",
+    )
+    .tuya_enum(
+        dp_id=21,
+        attribute_name="alarm_ringtone",
+        enum_class=NeoAlarmMelody32,
+        translation_key="alarm_ringtone",
+        fallback_name="Alarm ringtone",
+    )
+    .skip_configuration()
+    .add_to_registry()
+)
+
+
+# Tuya smart siren
+(
+    TuyaQuirkBuilder("_TZE204_k7mfgaen", "TS0601")
+    .applies_to("_TZE204_fncxk3ob", "TS0601")
+    .tuya_enum(
+        dp_id=1,
+        attribute_name="alarm_mode",
+        enum_class=TuyaSirenState,
+        translation_key="alarm_mode",
+        fallback_name="Alarm mode",
+    )
+    .tuya_enum(
+        dp_id=5,
+        attribute_name="alarm_volume",
+        enum_class=TuyaAlarmVolume,
+        translation_key="alarm_volume",
+        fallback_name="Alarm volume",
+    )
+    .tuya_binary_sensor(
+        dp_id=6,
+        attribute_name="charge_state",
+        translation_key="charge_state",
+        device_class=BinarySensorDeviceClass.BATTERY_CHARGING,
+        fallback_name="Charge state",
+    )
```

Comment:
I wonder if we need/want the `translation_key` for this, since it also has a device class.
I'll see if I can find something in HA where it's not used.

Reply:
```suggestion
        device_class=BinarySensorDeviceClass.BATTERY_CHARGING,
        fallback_name="Charging",
```